### ISSUE TITLE
Fix bug with aws-broker parameter group logic

### ISSUE THREAD

**Issue by markdboyd** [cloud-gov/aws-broker#282]
## Background

A customer encountered an error trying to use the `enable_pg_cron` feature when updating their database instance: https://cloud-gov-new.zendesk.com/agent/tickets/6455. Somehow the broker created a postgres14 family parameter group and is trying to attach it to the postgres12 database, which is failing.

We need to investigate the cause of the issue and update the broker to resolve it.

## Acceptance criteria

- [ ] Update broker to choose correct parameter group family
- [ ] Add unit tests to ensure parameter group family logic works correctly

**Comment by markdboyd** [comment]
Research so far:

- The logic to select the parameter group family for a database is here in the broker: https://github.com/cloud-gov/aws-broker/blob/master/services/rds/parameter_group.go#L103-L136
  - The logic branches based on whether the database service instance record loaded from the broker's database has `DbVersion` or not
- I confirmed that `DbVersion` is **not set** in the broker database for the database service that had the issue when upgrading
- When `DbVersion` is set, a request is made to AWS for `describe-db-engine-versions`. And I can confirm that command now returns `postgres14`:

    ```
    $ aws rds describe-db-engine-versions --engine postgres --default-only | jq '.DBEngineVersions[].DBParameterGroupFamily'
    "postgres14"
    ```

So I'm betting that something changed in AWS where the default parameter group family for Postgres is now `postgres14`, which explains why the broker tried to create that parameter group. And why it doesn't work for a database that is still PostgresSQL 12.

- There is code that should always set the `DbVersion` for new databases: https://github.com/cloud-gov/aws-broker/blame/master/services/rds/rdsinstance.go#L173-L178. However, this code was added in 2021, whereas the database was created in 2020.
